dnf5daemon-server/dbus/interfaces/org.rpm.dnf.v0.Goal.xml

```diff
@@ -94,6 +94,36 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
         <arg name="options" type="a{sv}" direction="in" />
     </method>
 
+
+    <!--
+        offline_transaction_pending:
+        @pending: boolean, true if there is a pending offline transaction
+
+        Check whether there is an offline transaction configured for the next reboot (i.e. checks the /system-update symlink existence).
+    -->
+    <method name="offline_transaction_pending">
+        <arg name="pending" type="b" direction="out" />
+    </method>
+
+
+    <!--
+        offline_transaction_clean:
+        @options: an array of key/value pairs
+        @transaction_cleaned: boolean, returns `false` if there was an error during the transaction cleanup. Returns `true` if the offline transaction was successfully cleaned or if no offline transaction was configured.
+
+        Clean the offline transaction configured for the next reboot by removing the /system-update symlink (see systemd.offline-updates(7)).
+
+        Following @options are supported:
+            - clean_datadir: boolean, default true
+                If true, all stored offline transaction data, including downloaded packages, will be removed along with the system-update symlink.
+
+        Unknown options are ignored.
+    -->
+    <method name="do_transaction">
+        <arg name="options" type="a{sv}" direction="in" />
+        <arg name="transaction_cleaned" type="b" direction="out" />
+    </method>
+
 </interface>
 
 </node>
```

Comment:
I'm not sure whether adding this to the `Goal` is the best place. For me, the Goal manages actual transactions, something ready to be started. These two look like a method not related to actual transactions.

Reply:
I agree it's not the best possible place. It's kind of loosely connected - the offline transactions are scheduled using `Goal::do_transaction(offline=true)`. But maybe introducing completely new `Offline` interface would be cleaner way.

Reply:
I did not think of a completely new interface, I thought some could be extended, but it seems none matches cleanly. The `Base` does only the `read_all_repos` and some signals, the `SessionManager` is not ideal either. I mean, you are right, adding the `Offline` interface looks like the best option.